tests/antares/services/local_services/output_service/test_output_local.py

```diff
@@ -0,0 +1,443 @@
+            areas_ids = params.get("areas_ids")
+            columns_names = params.get("columns_names")
+
+            output = setup_output(tmp_path, output_id)
+
+            df = output.aggregate_areas_mc_all(
+                query_file, frequency.value, areas_ids=areas_ids, columns_names=columns_names, mc_years=mc_years
+            )
+
+            resource_file = Path(ASSETS_DIR).joinpath(f"aggregate_areas_raw_data/{expected_result_filename}")
+
+            if not resource_file.exists() or resource_file.stat().st_size == 0:
+                df.to_csv(resource_file, sep="\t", index=False)
+
+            expected_df = pd.read_csv(resource_file, sep="\t", header=0).replace({np.nan: None})
+
+            for col in expected_df.columns:
+                expected_df[col] = expected_df[col].astype(df[col].dtype)
+
+            pd.testing.assert_frame_equal(df, expected_df)
+
+    def test_area_aggregate_mc_ind(self, tmp_path):
+        for params, expected_result_filename in AREAS_REQUESTS__IND:
+            output_id = params["output_id"]
+            frequency = Frequency(params["frequency"])
+            query_file = params.get("query_file")
+            mc_years = params.get("mc_years")
+            areas_ids = params.get("areas_ids")
+            columns_names = params.get("columns_names")
+
+            output = setup_output(tmp_path, output_id)
+
+            df = output.aggregate_areas_mc_ind(
+                query_file, frequency.value, areas_ids=areas_ids, columns_names=columns_names, mc_years=mc_years
+            )
+
+            resource_file = Path(ASSETS_DIR).joinpath(f"aggregate_areas_raw_data/{expected_result_filename}")
+
+            if not resource_file.exists() or resource_file.stat().st_size == 0:
+                df.to_csv(resource_file, sep="\t", index=False)
+
+            expected_df = pd.read_csv(resource_file, sep="\t", header=0).replace({np.nan: None})
+
+            for col in expected_df.columns:
+                expected_df[col] = expected_df[col].astype(df[col].dtype)
+
+            pd.testing.assert_frame_equal(df, expected_df)
+
+    def test_link_aggregate_mc_all(self, tmp_path):
+        for params, expected_result_filename in LINKS_REQUESTS__ALL:
+            output_id = params["output_id"]
+            frequency = Frequency(params["frequency"])
+            query_file = params.get("query_file")
+            mc_years = params.get("mc_years")
+            areas_ids = params.get("areas_ids")
+            columns_names = params.get("columns_names")
+
+            output = setup_output(tmp_path, output_id)
+
+            df = output.aggregate_links_mc_all(
+                query_file, frequency.value, areas_ids=areas_ids, columns_names=columns_names, mc_years=mc_years
+            )
+
+            resource_file = Path(ASSETS_DIR).joinpath(f"aggregate_links_raw_data/{expected_result_filename}")
+
+            if not resource_file.exists() or resource_file.stat().st_size == 0:
```

Comment:
Since this is used multiple times, and to make it more explicit why we're doing it, we can extract those 2 lines into a small function, for example:
```python
def update_expected_result_if_absent(path: Path, df: Dataframe) -> None:
    ...
```